libminifi/src/io/tls/TLSSocket.cpp

```diff
@@ -182,7 +182,15 @@ int16_t TLSSocket::initialize(bool blocking) {
     setNonBlocking();
   logger_->log_trace("Initializing TLSSocket %d", is_server);
   int16_t ret = context_->initialize(is_server);
-  Socket::initialize();
+
+  if (ret == 0) {
```

Comment:
This control flow is not very readable this way, with `ret` controlling what parts of the code run, sometimes with `== 0`, sometimes with `!= 0` and sometimes with `!ret`. Is there a reason we can't just log and `return -1` each time something fails? I think it would be more readable and would require less nested ifs.

Reply:
Fair point, simplified a bit.